numpyro/contrib/forecast/gt.py

```diff
@@ -0,0 +1,285 @@
+
+            if self.seasonality2 > self.seasonality:
+                s2_sm = numpyro.sample("s2_sm", dist.Uniform())
+                if self.generalized_seasonality:
+                    init_s2 = numpyro.sample("init_s2",
+                                             dist.Cauchy(0, data[:self.seasonality2] * 0.3))
+                    pow_season2 = numpyro.sample("pow_season2",
+                                                 dist.Beta(hypers["pow_season_alpha"],
+                                                           hypers["pow_season_beta"]))
+                else:
+                    init_s2 = numpyro.sample("init_s2", dist.Cauchy(0, 4))
+                    pow_season2 = None
+                exp_val, smoothed_innov_size = s2gt_scan(y, duration, coef_trend, pow_trend,
+                                                         lev_sm, innov_sm, innov_size_init,
+                                                         s_sm, init_s, pow_season,
+                                                         s2_sm, init_s2, pow_season2, llev_sm)
+            else:
+                exp_val, smoothed_innov_size = sgt_scan(y, duration, coef_trend, pow_trend,
+                                                        lev_sm, innov_sm, innov_size_init,
+                                                        s_sm, init_s, pow_season, llev_sm)
+        else:
+            loc_trend_fract = numpyro.sample("loc_trend_fract", dist.Uniform())
+            b_sm = numpyro.sample("b_sm", dist.Uniform())
+            b_init = numpyro.sample("b_init", dist.Cauchy(0, cauchy_sd))
+            exp_val, smoothed_innov_size = lgt_scan(y, duration, coef_trend, pow_trend,
+                                                    lev_sm, innov_sm, innov_size_init,
+                                                    loc_trend_fract, b_sm, b_init)
+
+        exp_val = exp_val + r[1:]
+        if duration > N:  # forecasting
+            exp_val = np.clip(exp_val[N - 1:], hypers["min_val"], hypers["max_val"])
+            if smoothed_innov_size is not None:
+                smoothed_innov_size = smoothed_innov_size[N - 1:]
+
+        nu = numpyro.sample("nu", dist.Uniform(hypers["min_nu"], hypers["max_nu"]))
+        sigma = numpyro.sample("sigma", dist.HalfCauchy(cauchy_sd))
+        offset_sigma = numpyro.sample("offset_sigma", dist.HalfCauchy(cauchy_sd)) \
+            + hypers["min_sigma"]
+        if self.use_smoothed_error:
+            omega = sigma * smoothed_innov_size + offset_sigma
+        else:
+            powx = numpyro.sample("powx", dist.Uniform())
+            omega = sigma * exp_val ** powx + offset_sigma
+
+        if covariates.shape[0] == N:  # training
+            numpyro.sample("y", dist.StudentT(nu, exp_val, omega), obs=data[1:])
+        else:  # forecasting
+            numpyro.sample("y", dist.StudentT(nu, exp_val, omega))
+
+
+# src: https://github.com/cbergmeir/Rlgt/blob/master/Rlgt/src/stan_files/LGT.stan
```

Comment:
The source file is GPL licensed. Can you find a non-GPL source for a similar model?

Reply:
Hmm, this is an important point... I'll learn more about the licence to make changes accordingly.

Reply:
Looking like there is no alternative unless we add a GPL license. I'll close this PR and think about other models which I can implement myself...